client/issue-lib.js

```diff
@@ -0,0 +1,144 @@
+(function() {
+'use strict';
+
+if (window.Issue) {
+  return;
+}
+
+// Colours are copied from:
+// https://www.google.com/design/spec/style/color.html#color-color-palette
+var _reviewLevelMetadata = {
+  'daily': {
+    query: {label: 'Update-Daily'},
+    outOfSLOColor: '#B71C1C',  // Red 900
+  },
+  'weekly': {
+    query: {label: 'Update-Weekly'},
+    outOfSLOColor: '#C62828',  // Red 800
+  },
+  'fortnightly': {
+    query: {label: 'Update-Fortnightly'},
+    outOfSLOColor: '#D32F2F',  // Red 700
+  },
+  'monthly': {
+    query: {label: 'Update-Monthly'},
+    outOfSLOColor: '#E53935',  // Red 600
+  },
+  'quarterly': {
+    query: {label: 'Update-Quarterly'},
+    outOfSLOColor: '#F44336',  // Red 500
+  },
+  'none (P0)': {
+    query: {label: 'Pri-0', '-has': 'update'},
+  },
+  'none (P1)': {
+    query: {label: 'Pri-1', '-has': 'update'},
+  },
+  'none (P2)': {
+    query: {label: 'Pri-2', '-has': 'update'},
+  },
+  'none (P3)': {
+    query: {label: 'Pri-3', '-has': 'update'},
+  },
+};
+var _defaultReviewLevel = 'none';
+var _inSLOColor = '#4CAF50';  // Green 500
+var _noSLOColor = '#757575';  // Grey 600
+
+// This class uses C style programming with JSON objects (representing issues)
+// and static methods.
+// This is to maintain the ability to clone the object in chromez-behaviours.js.
+class Issue {
+  static create({id, owner, summary, lastUpdatedString, labels}) {
+    var issue = {
+      id,
+      owner,
+      summary,
+      labels,
+      priority: undefined,
+      _lastUpdatedMS: Date.parse(lastUpdatedString),
+      _reviewLevel: _defaultReviewLevel,
+    };
+
+    for (var label of labels) {
+      if (label.substring(0, 4) == 'Pri-') {
+        issue.priority = Number(label.substring(4));
+      }
+      if (label.substring(0, 7) == 'Update-') {
+        var reviewLevel = label.substring(7).toLowerCase();
+        if (reviewLevel in _reviewLevelMetadata) {
+          issue._reviewLevel = reviewLevel;
+        }
+      }
+    }
+
+    return issue;
+  }
+
+  static daysSinceUpdate(issue) {
+    return Math.floor((Date.now() - issue._lastUpdatedMS) / (1000 * 60 * 60 * 24));
+  }
+
+  static reviewLevelWithBackoff(issue) {
+    var result = issue._reviewLevel;
+    if (result == _defaultReviewLevel) {
+      result += ' (P' + this.priority + ')';
```

Comment:
Oh, missed this before. this.priority --> issue.priority?

This code assumed that this.priority would be defined because all Monorail issues had a Pri-X label. I don't think this will hold for GitHub. That might be something to change in future patches, however.